start-service.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
```

Comment:
sorry, I haven't checked your previous working example and I just know that you automate this using bash script.
but I think, better we wrap these bash scripts with `Makefile`

So we can do something like this.
```bash
$ make start
$ make test
$ mate stop
```

Reply:
@Keda87 I have removed the shell script and wrote a Makefile. It works as expected. I have also updated the README.
- make build
![image](https://user-images.githubusercontent.com/67866345/112765430-d1ee1780-902a-11eb-857e-6a40c5d76157.png)
- make start
![image](https://user-images.githubusercontent.com/67866345/112765449-e3cfba80-902a-11eb-8189-40ead1ee5f79.png)
- make test and make stop
![image](https://user-images.githubusercontent.com/67866345/112765468-f518c700-902a-11eb-8366-a4af17d010ef.png)

Reply:
I'm trying in my local machine, but the DB seems does not create.

```bash
$  make start

Starting banking system service...
docker-compose up -d
Creating network "django-simple-banking-api_front-net" with the default driver
Creating network "django-simple-banking-api_back-net" with the default driver
Creating django-simple-banking-api_bank_db_1       ... done
Creating django-simple-banking-api_reverse_proxy_1 ... done
Recreating django-simple-banking-api_bank_api_1    ... done
Creating the bank database and performing migrations...
docker-compose exec bank_api python manage.py migrate
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 197, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/postgresql/base.py", line 185, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.8/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  database "db_banking" does not exist


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 15, in <module>
    execute_from_command_line(sys.argv)
```

Reply:
@Keda87 This was my first time writing a Makefile. I am sorry for the inconvenience. I have fixed the above issue.  The make command works for both of the following cases:

1. When db does not exist
![image](https://user-images.githubusercontent.com/67866345/112781583-c02b6500-9068-11eb-8f7e-cf6ec5081c65.png)

2. When db already exists
![image](https://user-images.githubusercontent.com/67866345/112781598-cc172700-9068-11eb-9d9a-0e2d5a967708.png)

Thanks for taking time and looking into this.

Reply:
it's okay, no worry.
actually, the changes are quite enough but need some little fix.

I think, `make start` will error if started for the first time.
because this following line executed in the beginning before all container started
```
docker_exists = $(shell docker-compose exec bank_db psql --username=postgres -tAc "SELECT 1 FROM pg_database WHERE datname='db_banking'")
```

<img width="1440" alt="Screen Shot 2021-03-29 at 10 23 02" src="https://user-images.githubusercontent.com/2397407/112783172-e48f3d80-9078-11eb-9727-8d98e13367e1.png">

Reply:
to reproduce, you can `make stop` then `make start` again

Reply:
@Keda87 If re-run the make start command again this issue doesn't exists. I have done everything I could to fix this in Makefile itself, but it's been unsuccessful. According to docs, this type of variable declaration should always re-evaluate everytime it's called. But, other than this I tried a bunch of stuff and all of this failed for the above case
1. Defining a function
2. Waiting / sleeping for sometime in case the db takes times to boot up ( issue persisted even with 25 sec)
3. Declaring a variable local to the target (this caused issues with normal behaviour itself)
4. Declaring a make command which evaluated the value of the definiton
5. Declaring a variable which  evaluates and stores the value of the definiton

However, I have 2 proposals which will work:
- if we use the same db "db_banking" in all env, then we can just define this in docker-compose.yml. It'll make sure to create the db, even if it doesn't exits
- if can write a shell script and call it from inside the Makefile target, you can still use the make commands and it'll be possible for me to do a lot more checks (say if not all containers are running, then cannot run tests), which seems to be very difficult to do in Makefile.

Reply:
Ohh okay, maybe we can keep our Makefile and move the DB creation within docker-compose.
I found this solution, have you tried this [one](https://stackoverflow.com/a/26599273/1936697)?

Reply:
Hi @Keda87. Thanks for providing the above reference. We can achieve what's given here like how we currently set postgres db password using **POSTGRES_PASSWORD** parameter for testing environment, I am referring to [this](https://github.com/Keda87/django-simple-banking-api/blob/master/docker-compose.yml#L37). 
Hence I referred to [this](https://hub.docker.com/_/postgres/) postgres docker official docs and found how we can do the same for username and DB creation on uping the docker container. I have made this change and pushed the changes for this. Could you please check this?

Reply:
To check the new changes, please don't forget to rebuilt the images using `make build` command. TIA.